firstvoices/backend/tests/test_apis/test_category_api.py

```diff
@@ -104,3 +101,155 @@ def test_detail_404(self):
+
+        assert response.status_code == 200
+        assert response_data["count"] == 0
+
+    @pytest.mark.django_db
+    def test_contains_phrase(self):
+        phrase_entry = DictionaryEntryFactory(
+            site=self.site, type=TypeOfDictionaryEntry.PHRASE
+        )
+        category_phrase = ParentCategoryFactory(site=self.site)
+        category_phrase.dictionary_entries.add(phrase_entry)
+
+        # Testing for PHRASE flag
+        response = self.client.get(
+            self.get_list_endpoint(
+                self.site.slug, query_kwargs={"contains": TypeOfDictionaryEntry.PHRASE}
+            )
+        )
+        response_data = json.loads(response.content)
+
+        assert response.status_code == 200
+        assert response_data["count"] > 0
+
+        assert response_data["results"][0] == {
+            "id": str(category_phrase.id),
+            "title": category_phrase.title,
+            "description": category_phrase.description,
+            "children": [],
+        }
+
+        # Testing for WORD flag, should not return anything
+        response = self.client.get(
+            self.get_list_endpoint(
+                self.site.slug, query_kwargs={"contains": TypeOfDictionaryEntry.WORD}
+            )
+        )
+        response_data = json.loads(response.content)
+
+        assert response.status_code == 200
+        assert response_data["count"] == 0
+
+    @pytest.mark.django_db
+    def test_contains_multiple(self):
+        word_entry = DictionaryEntryFactory(site=self.site)
+        phrase_entry = DictionaryEntryFactory(
+            site=self.site, type=TypeOfDictionaryEntry.PHRASE
+        )
+
+        category_word = ParentCategoryFactory(site=self.site)
+        category_phrase = ParentCategoryFactory(site=self.site)
+        category_both = ParentCategoryFactory(site=self.site)
+
+        category_word.dictionary_entries.add(word_entry)
+        category_phrase.dictionary_entries.add(phrase_entry)
+        category_both.dictionary_entries.add(word_entry, phrase_entry)
+
+        response = self.client.get(
+            self.get_list_endpoint(
+                self.site.slug,
+                query_kwargs={
+                    "contains": f"{TypeOfDictionaryEntry.WORD}|{TypeOfDictionaryEntry.PHRASE}"
+                },
+            )
+        )
+        response_data = json.loads(response.content)
+
+        assert response.status_code == 200
+        assert response_data["count"] == 3
+
+        ids_in_response = [
+            str(response_obj["id"]) for response_obj in response_data["results"]
+        ]
+        actual_ids = [
+            str(category_word.id),
+            str(category_phrase.id),
+            str(category_both.id),
+        ]
+
+        assert len(actual_ids) == len(ids_in_response)
+        # Checking all ids match up
+        difference = set(actual_ids) ^ set(ids_in_response)
+        assert not difference
+
+    @pytest.mark.django_db
+    def test_nested_categories(self):
```

Comment:
Is this meant to test that contains also applies to the child categories? If so I think we need to add one child that is a phrase, to prove it gets filtered out. And maybe add "contains" in the test name